alf/utils/tensor_utils.py

```diff
@@ -61,3 +63,22 @@ def explained_variance(ypred, y):
     y = y.view(-1)
     vary = torch.var(y, dim=0, unbiased=False)
     return 1 - torch.var(y - ypred, dim=0, unbiased=False) / (vary + 1e-30)
+
+
+def _to_tensor(data, dtype=None):
+    """Convert the data to a torch tensor.
+
+    Args: 
+        data (array like): data for the tensor. Can be a list, tuple, 
+            numpy ndarray, scalar, and other types.
+        dtype (torch.dtype): dtype of the converted tensors.
+
+    Returns:
+        A tensor of dtype
+    """
+    if not torch.is_tensor(data):
+        # as_tensor reuses the underlying data store of numpy array if possible.
+        data = torch.as_tensor(data, dtype=dtype).detach()
+        if alf.get_default_device() == "cuda":
+            data = data.cuda()
```

Comment:
I think `data.cuda()` is unnecessary because after we call `set_default_device()` the data will always be on the default device?

Reply:
updated.